xinference/core/utils.py

```diff
@@ -162,3 +162,28 @@ def parse_model_version(model_version: str, model_type: str) -> Tuple:
         return tuple(results)
     else:
         raise ValueError(f"Not supported model_type: {model_type}")
+
+
+def _get_nvidia_gpu_mem_info(gpu_id: int) -> Dict[str, float]:
+    from pynvml import nvmlDeviceGetHandleByIndex, nvmlDeviceGetMemoryInfo
+
+    handler = nvmlDeviceGetHandleByIndex(gpu_id)
+    mem_info = nvmlDeviceGetMemoryInfo(handler)
+    return {"total": mem_info.total, "used": mem_info.used, "free": mem_info.free}
+
+
+def get_nvidia_gpu_info() -> Dict:
+    try:
+        from pynvml import nvmlDeviceGetCount, nvmlInit, nvmlShutdown
+
+        nvmlInit()
+        device_count = nvmlDeviceGetCount()
```

Comment:
The `nvmlDeviceGetCount` may raise an exception: https://github.com/gpuopenanalytics/pynvml?tab=readme-ov-file#functions

It's better to call `nvmlShutdown` in finally.

Reply:
Fix